aws_advanced_python_wrapper/limitless_connection_plugin.py

```diff
@@ -0,0 +1,543 @@
+        return list_of_host_infos
+
+    def _create_host_info(self, result: Tuple[Any, Any], host_post_to_map: int) -> HostInfo:
+        host_name: str = result[0]
+        cpu: float = float(result[1])
+
+        weight: int = round(10 - (cpu * 10))
+        if weight < 1 or weight > 10:
+            weight = 1
+            logger.debug("LimitlessRouterMonitor.InvalidRouterLoad", host_name, cpu)
+
+        return HostInfo(host_name, host_post_to_map, weight=weight, host_id=host_name)
+
+
+class LimitlessConnectionContext:
+
+    def __init__(self,
+                 host_info: HostInfo,
+                 props: Properties,
+                 connection: Optional[Connection],
+                 connect_func: Callable,
+                 limitless_routers: List[HostInfo],
+                 connection_plugin: LimitlessConnectionPlugin) -> None:
+        self._host_info = host_info
+        self._props = props
+        self._connection = connection
+        self._connect_func = connect_func
+        self._limitless_routers = limitless_routers
+        self._connection_plugin = connection_plugin
+
+    def get_host_info(self):
+        return self._host_info
+
+    def get_props(self):
+        return self._props
+
+    def get_connection(self):
+        return self._connection
+
+    def set_connection(self, connection: Connection):
+        if self._connection is not None and self._connection != connection:
+            try:
+                self._connection.close()
+            except Exception:
+                pass
+
+        self._connection = connection
+
+    def get_connect_func(self) -> Callable:
+        return self._connect_func
+
+    def get_limitless_routers(self):
+        return self._limitless_routers
+
+    def set_limitless_routers(self, limitless_routers: List[HostInfo]):
+        self._limitless_routers = limitless_routers
+
+    def get_connection_plugin(self):
+        return self._connection_plugin
+
+    def is_any_routers_available(self):
+        for router in self.get_limitless_routers():
+            if router.get_availability() == HostAvailability.AVAILABLE:
+                return True
+        return False
+
+
+class LimitlessRouterService:
+    _CACHE_CLEANUP_NS: int = 6 * 10 ^ 10  # 1 minute
+    _limitless_router_cache: ClassVar[SlidingExpirationCacheWithCleanupThread[str, List[HostInfo]]] = \
+        SlidingExpirationCacheWithCleanupThread(_CACHE_CLEANUP_NS,
+                                                should_dispose_func=lambda x: True,
+                                                item_disposal_func=lambda x: {})
+
+    _limitless_router_monitor: ClassVar[SlidingExpirationCacheWithCleanupThread[str, LimitlessRouterMonitor]] = \
+        SlidingExpirationCacheWithCleanupThread(_CACHE_CLEANUP_NS,
+                                                should_dispose_func=lambda monitor: True,
+                                                item_disposal_func=lambda monitor: monitor.close())
```

Comment:
nit: for the lambdas that don't do anything I think you can just skip passing the arg since the SlidingExpirationCache class provides default values of None for them and should handle the default behavior in the same way
```suggestion
    _limitless_router_cache: ClassVar[SlidingExpirationCacheWithCleanupThread[str, List[HostInfo]]] = \
        SlidingExpirationCacheWithCleanupThread(_CACHE_CLEANUP_NS)

    _limitless_router_monitor: ClassVar[SlidingExpirationCacheWithCleanupThread[str, LimitlessRouterMonitor]] = \
        SlidingExpirationCacheWithCleanupThread(_CACHE_CLEANUP_NS, 
                                                item_disposal_func=lambda monitor: monitor.close())
```